connector/exceptionsconnector/connector.go

```diff
@@ -0,0 +1,348 @@
+func (p *connectorImp) ConsumeTraces(_ context.Context, traces ptrace.Traces) error {
+	// Forward trace data unmodified and propagate both metrics and trace pipeline errors, if any.
+	p.lock.Lock()
+	p.aggregateExceptionMetrics(traces)
+	p.lock.Unlock()
+	return nil
+}
+
+func (p *connectorImp) exportMetrics(ctx context.Context) error {
+	p.lock.Lock()
+	m, err := p.buildExceptionMetrics()
+	p.resetAccumulatedMetrics()
+
+	// This component no longer needs to read the metrics once built, so it is safe to unlock.
+	p.lock.Unlock()
+
+	if err != nil {
+		return err
+	}
+
+	if err = p.metricsConsumer.ConsumeMetrics(ctx, m); err != nil {
+		p.logger.Error("Failed ConsumeMetrics", zap.Error(err))
+		return err
+	}
+	return nil
+}
+
+// getDimensionsByMetricKey gets dimensions from `metricKeyToDimensions` cache.
+func (p *connectorImp) getDimensionsByMetricKey(k metricKey) (pcommon.Map, error) {
+	if attributeMap, ok := p.metricKeyToDimensions.Get(k); ok {
+		return attributeMap, nil
+	}
+	return pcommon.Map{}, fmt.Errorf("value not found in metricKeyToDimensions cache by key %q", k)
+}
+
+// aggregateExceptionMetrics aggregates data from spans to generate metrics.
+func (p *connectorImp) aggregateExceptionMetrics(traces ptrace.Traces) {
+	for i := 0; i < traces.ResourceSpans().Len(); i++ {
+		rspans := traces.ResourceSpans().At(i)
+		resourceAttr := rspans.Resource().Attributes()
+		serviceAttr, ok := resourceAttr.Get(conventions.AttributeServiceName)
+		if !ok {
+			continue
+		}
+		serviceName := serviceAttr.Str()
+		ilsSlice := rspans.ScopeSpans()
+		for j := 0; j < ilsSlice.Len(); j++ {
+			ils := ilsSlice.At(j)
+			spans := ils.Spans()
+			for k := 0; k < spans.Len(); k++ {
+				span := spans.At(k)
+				for l := 0; l < span.Events().Len(); l++ {
+					event := span.Events().At(l)
+					if event.Name() == "exception" {
+						attr := event.Attributes()
+
+						// Always reset the buffer before re-using.
+						p.keyBuf.Reset()
+						buildKey(p.keyBuf, serviceName, span, p.dimensions, attr)
+						key := metricKey(p.keyBuf.String())
+						p.cache(serviceName, span, key, attr)
+						p.updateException(key, span.TraceID(), span.SpanID())
+					}
+				}
+			}
+		}
+	}
+}
+
+// buildExceptionMetrics collects the computed raw metrics data, builds the metrics object and
+// writes the raw metrics data into the metrics object.
+func (p *connectorImp) buildExceptionMetrics() (pmetric.Metrics, error) {
+	m := pmetric.NewMetrics()
+	ilm := m.ResourceMetrics().AppendEmpty().ScopeMetrics().AppendEmpty()
+	ilm.Scope().SetName("exceptionsconnector")
+
+	if err := p.collectExceptions(ilm); err != nil {
+		return pmetric.Metrics{}, err
+	}
+
+	p.metricKeyToDimensions.RemoveEvictedItems()
+
+	return m, nil
+}
+
+// collectExceptions collects the exception metrics data and writes it into the metrics object.
+func (p *connectorImp) collectExceptions(ilm pmetric.ScopeMetrics) error {
+	mCalls := ilm.Metrics().AppendEmpty()
+	mCalls.SetName("exceptions_total")
```

Comment:
I think `exceptions_total` should be `exceptions`, since total will be added to name automatically